src/Storage/Entity/Entity.php

```diff
@@ -73,8 +77,36 @@ public function setId($id)
         $this->id = $id;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function __toString()
     {
         return (string) $this->getId();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toArray()
+    {
+        $data = [];
+        foreach ($this as $k => $v) {
```

Comment:
(Ignore the visual references to 3.3, it is just my 3.3 worktree checked out with this branch)

![entity backtrace](https://user-images.githubusercontent.com/1427081/27320754-ddf1b2ac-5597-11e7-9825-08d071d5babf.png)